### ISSUE TITLE
[VSCode extension] Typing dot autocompletes in unintuitive ways

### ISSUE THREAD

**Issue by anthonyvdotbe** [apache/netbeans#5130]
### Apache NetBeans version

Apache NetBeans 16

### What happened

Consider the following code:

```java
class Bar {

    static void bar() {
        class Foo { static void foo() {} }
        // cursor here
    }
}
```

Now type `Foo`. At this point NetBeans provides suggestions:

![image](https://user-images.githubusercontent.com/15714253/208899877-c3fc1825-24cd-46ee-a517-250a23d14f53.png)

Then type a dot. Now NetBeans autocompletes `Foo.` to `ForLoopTree.` and gives suggestions for it.

What I expected to happen, is for NetBeans to just add the dot, and give suggestions for the class `Foo` defined in the line above:

![image](https://user-images.githubusercontent.com/15714253/208900562-33ec3e84-1442-47ca-b2c7-bc6a7080b20e.png)

In other words: when what's before the dot is valid in the current context, typing a dot should never (unless there's sensible exceptions I'm missing) autocomplete. When I want to autocomplete to a type, I can easily press Tab or Enter to autocomplete first, and then type the dot.

### How to reproduce

As described in `What happened`.

### Did this work correctly in an earlier version?

No / Don't know

### Operating System

Windows 11

### JDK

openjdk full version "20-ea+22-1594"

### Apache NetBeans packaging

Apache VSNetBeans for VSCode

### Are you willing to submit a pull request?

Yes

### Code of Conduct

Yes

**Comment by lahodaj** [comment]
FWIW, I think the real issue here is that the code completion does not include `Foo`, rather than the action done on pressing the dot. I think I have a fair idea on why is that. Much less an idea on how to fix, yet.

**Comment by bowbahdoe** [comment]
Hard disagree. In this case the dependency is missing because of an editor thing, but it could easily be misconfiguration. The dot causing an autofill is surprising behavior every time it happens

**Comment by lahodaj** [comment]
FWIW, there is:
https://github.com/apache/netbeans/pull/8515

At the same time, I guess I don't quite understand why having typed `System.er`, invoking code completion, selecting `err` in the code completion and pressing `.` should lead to `System.er.` rather than `System.err.`. But as long as there's a setting that would allow me to have the behavior that makes more sense to me (i.e. `System.err.`), there can also be the other behavior.

**Comment by bowbahdoe** [comment]
I bumped into this last night actually. I was helping someone understand their end of semester "translate this python program into Java" assignment starting with the anonymous main class version since.

```java
void main() {
    String accountNumber = IO.readln("Enter the account number ");
    IO.println("Account number " + accountNumber);
}
```
is pretty directly explainable as a parallel to

```java
account_number = input("Enter the account number ")
println(f"Account number {account_number}")
```

But even though the extension was fine to enable the preview feature that allows `void main()` it didn't know about the IO class for whatever reason.

Every time I typed `IO.` it would turn into `IOError.` I needed to first write `.readln` then use the arrows to get to the left of the `.` and type `IO` again.

> invoking code completion, selecting err in the code completion and pressing . should lead to System.er.

I think what you are missing is that there is no "selecting err in the code completion" step. All the actions you took were the same as if you never looked at the completion.

**Comment by anthonyvdotbe** [comment]
Isn't this what Jan already mentioned [above](https://github.com/apache/netbeans/issues/5130#issuecomment-2737658296)? That sometimes the obvious completion (the class `Foo` in my OP, the class `IO` in your example) is missing from the list?

**Comment by bowbahdoe** [comment]
Right - but i'm saying that it being missing from the list is a situation to be accounted for. It can happen if you just forget to declare a dependency, its not always something fixable by making the plugin be better about having the class in the list.

When this scenario that is common happens, the experience is jarring

**Comment by anthonyvdotbe** [comment]
Another example:

I already have:

> import java.util.concurrent.StructuredTaskScope.Subtask.State;

and `subtask` is of type `StructuredTaskScope.Subtask<T>`. Yet when I type the dot, I get `Thread.State.`

<img width="1400" height="350" alt="Image" src="https://github.com/user-attachments/assets/797e704d-f0f4-4e10-bff9-0611b890a6e0" />
